src/frontend/src/components/MobileHeader/MobileHeader.jsx

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+import { Link } from 'gatsby';
+import { makeStyles } from '@material-ui/core/styles';
+import {
+  AppBar,
+  Button,
+  Toolbar,
+  Typography,
+  IconButton,
+  List,
+  ListItem,
+  Drawer,
+  Divider,
+} from '@material-ui/core';
+import MenuIcon from '@material-ui/icons/Menu';
+import SearchIcon from '@material-ui/icons/Search';
+
+import useSiteMetadata from '../../hooks/use-site-metadata';
+
+import Login from '../Login';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    flexGrow: 1,
+    backgroundColor: '#242424',
+  },
+  toolbar: theme.mixins.toolbar,
+  menuIcon: {
+    fontSize: '2.5rem',
+  },
+  searchIcon: {
+    fontSize: '2.5rem',
+    color: 'white',
+  },
+  title: {
+    flexGrow: 1,
+    marginLeft: '1rem',
+    color: '#a4d4ff',
+  },
+  links: {
+    color: 'white',
+    fontFamily: 'Roboto, sans-serif',
+    textDecoration: 'none',
+    fontSize: '1.5rem',
+    margin: '0 0.5rem 0 0.5rem',
+  },
+  button: {
+    float: 'right',
+    margin: '0 0.5rem 0 0.5rem',
+  },
+  list: {
+    width: 250,
+  },
+  paper: {
+    background: '#242424',
+  },
+  line: {
+    backgroundColor: '#525252',
+  },
+  item: {
+    textAlign: 'center',
+  },
+}));
+
+const Header = () => {
+  const { title } = useSiteMetadata();
+  const classes = useStyles();
+  const [state, setState] = React.useState({
+    right: false,
+  });
+
+  const toggleDrawer = (side, open) => (event) => {
+    if (event.type === 'keydown' && (event.key === 'Tab' || event.key === 'Shift')) {
+      return;
+    }
+
+    setState({ ...state, [side]: open });
+  };
+
+  const sideList = (side) => (
+    <div
+      className={classes.list}
+      role="presentation"
+      onClick={toggleDrawer(side, false)}
+      onKeyDown={toggleDrawer(side, false)}
+    >
+      <List>
+        <ListItem className={classes.item}>
```

Comment:
Small nit: 
Using HOME as an example here, this PR doesn't make the entire ListItem clickable, I would have to click on the HOME text to get to where I wanna go. If I click outside of the HOME but still inside of the ListItem, it closes the drawer but I'm still on the same page. This gives me the impression that the button doesn't work.

I suggest `<ListItem button component={Link} className={classes.item}>` for all ListItems, this will allow the entire ListItem to be clickable for better user experience :D